src/pages/CoinList/index.js

```diff
@@ -275,14 +309,15 @@ class CoinList extends React.Component {
             })}
           </>
         )}
-        {this.state.coinListLength === 0 && this.state.showFavorites && (
+        {!favoriteCoinsLength() && this.state.showFavorites && (
           <EmptyFavoriteList />
         )}
-        {this.state.isLoading && (
-          <div>
-            <SkeletonCoinList coinsPerPage={this.state.coinsPerPage} />
-          </div>
-        )}
+        {this.state.isLoading &&
+          (this.state.showFavorites && favoriteCoinsLength() < 1 ? (
+            <SkeletonCoinList coinsPerPage={favoriteCoinsLength()} />
+          ) : (
+            <SkeletonCoinList coinsPerPage={this.getScreenWidth()} />
+          ))}
```

Comment:
this block looks a bit messy, can you make a few variables out of all these conditions?